hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/LeanWriteStatus.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client;
+
+/**
+ * Lean version of {@link WriteStatus} where all additional stats like rli, col sats etc are trimmed off.
+ */
+public class LeanWriteStatus extends WriteStatus {
```

Comment:
Some thoughts here:
1. Does it make more sense to have the WriteStatus extend LeanWriteStatus?
2. Is there another name we should use to make it more clear when to use one class as compared to other? The WriteStatus is now containing more metadata whereas the LeanStatus is meant to have the actual final status of the write. If we go this route, you can also make the data required for the final result a member of the object that has the metadata as well. This `has a` instead of `is a` style relationship may make more sense.

Reply:
thanks. 
I def wanted to fix this properly. 
based on my thoughts and your suggestion, here is what I am thinking: 
Lets not add any new classes. We will have just 1 class named WriteStatus. 
we can add additional apis to WriteStatus class to drop these additional metadata. Alternatively, we could add add methods called .getFinalWriteStatus which will create another WriteStatus object, but it will incur the cost of creating new object. So, inclining towards adding a method named `.dropMetadataStats()" 

So, in the map call where we convert WriteStatus -> LeanWriteStatus as per this patch, we will just do 
```
writeStatus -> { 

writeStatus.dropMetadataStats()
return writeStatus; 
}
```

Reply:
I think this will be very confusing going forward. Developers will need to trace the code to understand which fields are meant to be set a different stages. The overhead of creating a new object is not that high when dealing with the metadata which is a much smaller scale than per record.